src/font/font_options.rs

```diff
@@ -0,0 +1,115 @@
+use glib::translate::*;
+use std::cmp::PartialEq;
+use ffi;
+
+use ffi::enums::{
+    Antialias,
+    SubpixelOrder,
+    HintStyle,
+    HintMetrics,
+};
+use ffi::cairo_font_options_t;
+
+glib_wrapper! {
+    pub struct FontOptions(Boxed<ffi::cairo_font_options_t>);
+
+    match fn {
+        copy => |ptr| ffi::cairo_font_options_copy(ptr),
+        free => |ptr| ffi::cairo_font_options_destroy(ptr),
+    }
+}
+
+impl FontOptions {
+    pub fn new() -> FontOptions {
+        let mut font_options: FontOptions = unsafe {
+            from_glib_full(ffi::cairo_font_options_create())
+        };
+        font_options.ensure_status();
+        font_options
+    }
+
+    #[doc(hidden)]
+    pub fn get_ptr(&self) -> *const cairo_font_options_t {
+        self.to_glib_none().0
+    }
+
+    #[doc(hidden)]
+    pub fn get_ptr_mut(&mut self) -> *mut cairo_font_options_t {
+        self.to_glib_none_mut().0
+    }
+
+    pub fn ensure_status(&mut self) {
+        let status = unsafe {
+            ffi::cairo_font_options_status(self.get_ptr_mut())
+        };
+        status.ensure_valid()
+    }
+
+    pub fn merge(&mut self, other: &mut FontOptions) {
+        unsafe {
+            ffi::cairo_font_options_merge(self.get_ptr_mut(), other.get_ptr_mut())
+        }
+    }
+
+    pub fn hash(&self) -> u64{
+        unsafe {
+            ffi::cairo_font_options_hash(self.get_ptr()) as u64
+        }
+    }
+
+    pub fn set_antialias(&mut self, antialias: Antialias) {
+        unsafe {
+            ffi::cairo_font_options_set_antialias(self.get_ptr_mut(), antialias)
```

Comment:
Since `FontOptions` is a `Boxed` object, and not `Shared`, I thought this might be the proper way to do it. And gnome docs has meticulously marked `get_*` pointers as `const` for `font_options` unlike other objects that I found.